### ISSUE TITLE
Exception during render job java.lang.ArrayIndexOutOfBoundsException

### ISSUE THREAD

**Issue by LeapingDolphin23** [webbukkit/dynmap#2067]
I have been getting this error when I try to load my Resource Pack up and run a fullrender. I am using Flows HD x64 1.11 (there is no lower version such as x32 or x16). Is there any way to fix this issue. It looks like it has some issue with the Shulker?
```
[17:00:45 INFO]: [dynmap] Loading resource pack Flows
[17:00:47 ERROR]: [dynmap] Exception during render job: world=world, map=org.dynmap.hdmap.HDMap@5d6a3a7e
[17:00:47 WARN]: java.lang.ArrayIndexOutOfBoundsException
[17:00:47 WARN]:        at java.lang.System.arraycopy(Native Method)
[17:00:47 WARN]:        at org.dynmap.hdmap.TexturePack.copySubimageFromImage(TexturePack.java:878)
[17:00:47 WARN]:        at org.dynmap.hdmap.TexturePack.makeShulkerSideImage(TexturePack.java:1074)
[17:00:47 WARN]:        at org.dynmap.hdmap.TexturePack.patchShulkerImages(TexturePack.java:1104)
[17:00:47 WARN]:        at org.dynmap.hdmap.TexturePack.processDynamicImage(TexturePack.java:1344)
[17:00:47 WARN]:        at org.dynmap.hdmap.TexturePack.<init>(TexturePack.java:852)
[17:00:47 WARN]:        at org.dynmap.hdmap.TexturePack.getTexturePack(TexturePack.java:683)
[17:00:47 WARN]:        at org.dynmap.hdmap.TexturePackHDShader.getTexturePack(TexturePackHDShader.java:57)
[17:00:47 WARN]:        at org.dynmap.hdmap.TexturePackHDShader.access$000(TexturePackHDShader.java:22)
[17:00:47 WARN]:        at org.dynmap.hdmap.TexturePackHDShader$ShaderState.<init>(TexturePackHDShader.java:128)
[17:00:47 WARN]:        at org.dynmap.hdmap.TexturePackHDShader.getStateInstance(TexturePackHDShader.java:323)
[17:00:47 WARN]:        at org.dynmap.hdmap.HDMapManager.getShaderStateForTile(HDMapManager.java:136)
[17:00:47 WARN]:        at org.dynmap.hdmap.IsoHDPerspective.render(IsoHDPerspective.java:1088)
[17:00:47 WARN]:        at org.dynmap.hdmap.HDMapTile.render(HDMapTile.java:79)
[17:00:47 WARN]:        at org.dynmap.MapManager$FullWorldRenderState.processTile(MapManager.java:708)
[17:00:47 WARN]:        at org.dynmap.MapManager$FullWorldRenderState.run(MapManager.java:641)
[17:00:47 WARN]:        at org.dynmap.MapManager$DynmapScheduledThreadPoolExecutor$1.run(MapManager.java:190)
[17:00:47 WARN]:        at org.dynmap.MapManager$DynmapScheduledThreadPoolExecutor$2.run(MapManager.java:206)
[17:00:47 WARN]:        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
[17:00:47 WARN]:        at java.util.concurrent.FutureTask.run(Unknown Source)
[17:00:47 WARN]:        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(Unknown Source)
[17:00:47 WARN]:        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
[17:00:47 WARN]:        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
[17:00:47 WARN]:        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
[17:00:47 WARN]:        at java.lang.Thread.run(Unknown Source)
```

**Comment by LeapingDolphin23** [comment]
Edit: Downgraded to Flow's HD 1.10 and surprisingly, the render started normally without any problems, and the pack appeared on the Dynmap. So I guess there was some incompatibility with 1,11 resource packs?

**Comment by wioxjk** [comment]
May I see your custom-shaders.txt?

**Comment by LeapingDolphin23** [comment]
Here it is:
`#
# This file contains default standard shader definitions.  The contents of this file are replaced and updated
# during upgrades, so new or updated shader definitions should be done in the custom-shaders.txt file
#
shaders:
  # Default shader - used if shader not specified or invalid
  - class: org.dynmap.hdmap.TexturePackHDShader
    name: default
    texturepack: Flows

  # Color scheme based shaders
  - class: org.dynmap.hdmap.DefaultHDShader
    name: defaultscheme
    colorscheme: default

  - class: org.dynmap.hdmap.DefaultHDShader
    name: ovocean
    colorscheme: ovocean
   
  - class: org.dynmap.hdmap.DefaultHDShader
    name: flames
    colorscheme: flames

  - class: org.dynmap.hdmap.DefaultHDShader
    name: sk89q
    colorscheme: sk89q
   
  - class: org.dynmap.hdmap.DefaultHDShader
    name: amidst
    biomecolored: biome
    colorscheme: amidst

  # Biome based shaders
  - class: org.dynmap.hdmap.DefaultHDShader
    name: biome
    biomecolored: biome
    
  - class: org.dynmap.hdmap.DefaultHDShader
    name: temperature
    biomecolored: temperature
    
  - class: org.dynmap.hdmap.DefaultHDShader
    name: rainfall
    biomecolored: rainfall
    
  # No transparency default color scheme shader
  - class: org.dynmap.hdmap.DefaultHDShader
    name: no_transparency
    colorscheme: default
    transparency: false
  
  # Cave view shader
  - class: org.dynmap.hdmap.CaveHDShader
    name: cave

  # Cave view shader - only lit caves
  - class: org.dynmap.hdmap.CaveHDShader
    name: lit-cave
    onlyiflit: true

  # Cave view shader - hide various soft blocks
  - class: org.dynmap.hdmap.CaveHDShader
    name: cave-noplants
    hiddenids: [ 6, 17, 18, 31, 32, 37, 38, 39, 40, 50, 55, 78, 81, 83, 86, 99, 100, 103, 104, 105, 111, 115 ]
    
  # Texture pack based shader for standard Minecraft textures
  - class: org.dynmap.hdmap.TexturePackHDShader
    name: stdtexture
    texturepack: Flows

  # Standard texture cave view shader
  - class: org.dynmap.hdmap.TexturePackHDCaveShader
    name: stdtexture-cave
    texturepack: Flows
    max-sky-light: 0
    min-emitted-light: 1
    
  # Texture pack based shader for standard Minecraft textures, without biome-tinted grass/leaves
  - class: org.dynmap.hdmap.TexturePackHDShader
    name: stdtexture-nobiome
    texturepack: Flows
    biomeshaded: false

  # Texture pack based shader for standard Minecraft textures, with grid for region file borders (512x512)
  - class: org.dynmap.hdmap.TexturePackHDShader
    name: stdtexture-mcr-grid
    texturepack: Flows
    grid-scale: 512
    
  # Topological map shader    
  - class: org.dynmap.hdmap.TopoHDShader
    name: topo
    color127: "#FFFFFF"
    color111: "#8B4513"
    color95: "#D2B48C"
    color79: "#FFFF00"
    color63: "#008000"
    color47: "#228B22"
    color31: "#104010"
    color15: "#6B8E23"
    color0: "#696969"
    linecolor: "#000000"
    watercolor: "#0000FF"
    wateralpha: 1.0

  # Topological map shader (spread over 256 height)
  - class: org.dynmap.hdmap.TopoHDShader
    name: topo256
    color255: "#FFFFFF"
    color222: "#8B4513"
    color190: "#D2B48C"
    color158: "#FFFF00"
    color126: "#008000"
    color94: "#228B22"
    color62: "#104010"
    color30: "#6B8E23"
    color0: "#696969"
    linecolor: "#000000"
    watercolor: "#0000FF"
    wateralpha: 1.0

  # Topological map shader (hide plants and other soft features)    
  - class: org.dynmap.hdmap.TopoHDShader
    name: topo-noplants
    color127: "#FFFFFF"
    color111: "#8B4513"
    color95: "#D2B48C"
    color79: "#FFFF00"
    color63: "#008000"
    color47: "#228B22"
    color31: "#104010"
    color15: "#6B8E23"
    color0: "#696969"
    linecolor: "#000000"
    watercolor: "#0000FF"
    wateralpha: 1.0
    hiddenids: [ 6, 17, 18, 31, 32, 37, 38, 39, 40, 50, 55, 78, 81, 83, 86, 99, 100, 103, 104, 105, 111, 115 ]
 
   # Topological map shader (hide plants and other soft features)    
  - class: org.dynmap.hdmap.TopoHDShader
    name: topo256-noplants
    color255: "#FFFFFF"
    color222: "#8B4513"
    color190: "#D2B48C"
    color158: "#FFFF00"
    color126: "#008000"
    color94: "#228B22"
    color62: "#104010"
    color30: "#6B8E23"
    color0: "#696969"
    linecolor: "#000000"
    watercolor: "#0000FF"
    wateralpha: 1.0
    hiddenids: [ 6, 17, 18, 31, 32, 37, 38, 39, 40, 50, 55, 78, 81, 83, 86, 99, 100, 103, 104, 105, 111, 115 ]
 
  # Alternate topological map shader (thanks to TerraNetworkOrg!)
  - class: org.dynmap.hdmap.TopoHDShader
    name: TerraNetworkOrgTopo
    color127: "#4F2101"
    color120: "#8B4513" 
    color108: "#D2B48C"
    color96: "#C99653"
    color84: "#CB9C0B"
    color72: "#B1B62E"
    color60: "#0F8D0F"
    color48: "#14703B"
    color36: "#106343"
    color24: "#0E5347"
    color12: "#939393"
    color0: "#696969"
    linecolor: "#000000"
    watercolor: "#1F1FEA"
    wateralpha: 1.0
    
  # Inhabited time map shader    
  - class: org.dynmap.hdmap.InhabitedHDShader
    name: inhabited
    color10: "#0000FF"
    color30: "#00FFFF"
    color100: "#008000"
    color300: "#FFFF00"
    color1000: "#FF8000"
    color3000: "#FF0000"
`
I just copied the existing shaders.txt and pasted the contents. The only changes I made was to change the Resource Pack to Flows. This is the configuration used for both the 1.10 and 1.11 resource packs (they were both named Flows).

**Comment by patrix87** [comment]
I'm also having the same issue with 1.11 Dokucraft Light.

If I extract the texture folder of the pack it "works" but a lot of textures are missing.

**Comment by SimonOrJ** [comment]
The problem roots from somehow malformed shulker box rendering code.  Check #2074; the issues are related.

**Comment by FrozenBeard** [comment]
Im also using Dokucraft light and having these issues, what I don't get is I even backdated to the Dokucraft Light 1.10 and I get the same issues.

**Comment by CloakedAlien** [comment]
Can confirm I'm also having this. Using A dokucraft-variation =/

**Comment by FedUpWith-Tech** [comment]
fixed in #2109
